x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/ClientVersion.java

```diff
@@ -87,7 +88,11 @@ static Manifest getManifest(URL url) throws IOException {
             if (url.getProtocol().equals("jar")) {
                 JarURLConnection jarConn = (JarURLConnection) conn;
                 if (jarConn.getEntryName() == null) { // the URL points to a JAR file
-                    return jarConn.getManifest(); // in case of a fat JAR, this would return the outermost JAR's manifest
+                    Manifest manifest = jarConn.getManifest(); // in case of a fat JAR, this would return the outermost JAR's manifest
+                    try {
+                        jarConn.getJarFile().close(); // prevent locked file errors in Windows. The Manifest has been read by now
+                    } catch (IOException e) {/* ignore any closing failure */}
```

Comment:
Maybe we should log here, since this is not only for testing.

Reply:
We could, but where: the CLI client uses `java.util.logging`, but JDBC uses it's own logging. Would hooking into either just for this warning - since it's not really an error - be justified?

Reply:
My bad, sorry, misread that the return is within the catch block.

Reply:
Probably not I guess...